### PR TITLE
Us537403 profiling sdk

### PR DESCRIPTION
**Pull request by Ruqiya-Kaladgi**
Please review

### CODE REVIEW

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+            scenarioInfo.setBenchmark(avg);
+        }
+        Log.d(TAG, "Benchmark = " + avg + "s");
+        assertTrue("Taken more than " +scenarioInfo.getBenchmark() +" time to execute", avg <= scenarioInfo.getBenchmark());
+
+    }
+
+    private static void printUpdatedBenchmark() {
+
+        Gson gson = new GsonBuilder().create();
```
Comment (Arkariang) [inline mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: for writing the result of the benchmark take a look into this method android.content.Context#openFileOutput

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+                        @Override
+                        public void onError(Throwable e) {
+
+                        }
+                    });
+                }
+            });
+
+            MAS.invoke(request,new MASCallback<MASResponse<JSONObject>>()
+            {
```
Comment (Arkariang) [inline mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: Not important, but be careful with the code style

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+            printUpdatedBenchmark();
+        }
+
+    }
+
+
+
+    @Test
+    @TestId(8)
+    public void startTest() throws URISyntaxException, ExecutionException, InterruptedException {
```
Comment (Arkariang) [inline mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: remove unused exceptions

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+        if (MASDevice.getCurrentDevice() != null && MASDevice.getCurrentDevice().isRegistered()) {
+            MASDevice.getCurrentDevice().deregister(null);
+        }
+        MAS.processPendingRequests();
+        MAS.stop();
+
+    }
+
+    @AfterClass
+    public static void compelete(){
```
Comment (Arkariang) [inline mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: typo "complete"

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.io.StringWriter;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Paths;
```
Comment (Arkariang) [inline mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: remove unused imports

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java (hunk 1 above)
Comment (Ruqiya-Kaladgi) [reply mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: This API will not work for our requirement of having a file readable and writable programmatically and manually.

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java (hunk 2 above)
Comment (Ruqiya-Kaladgi) [reply mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: Done

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java (hunk 3 above)
Comment (Ruqiya-Kaladgi) [reply mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: Done

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java (hunk 4 above)
Comment (Ruqiya-Kaladgi) [reply mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: Done

File: mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java (hunk 5 above)
Comment (Ruqiya-Kaladgi) [reply mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java]: Done

**Pull request merged**